packages/react-reconciler/src/ReactFiberScheduler.new.js

```diff
@@ -1404,6 +1404,15 @@ function commitRootImpl(root, expirationTime) {
     rootWithPendingPassiveEffects = root;
     pendingPassiveEffectsExpirationTime = expirationTime;
   } else {
+    // We are done with the effect chain at this point so let's clear the
+    // nextEffect pointers to assist with GC. If we have passive effects, we'll
+    // clear this in flushPassiveEffects.
+    nextEffect = firstEffect;
+    while (nextEffect !== null) {
+      const nextNextEffect = nextEffect.nextEffect;
+      nextEffect.nextEffect = null;
```

Comment:
I think this part of the fix makes sense because otherwise if this node (nextEffect) never gets any updates ever again, it'll now have a reference to a node that might later get deleted. Should be pretty easy to come up with a reduced test case for this.

Reply:
I was able to come up with an isolated repro. https://github.com/paulshen/react-next-effect-leak/blob/master/src/App.js Our belief was correct that `nextEffect` pointers can point to zombie nodes. Here is a running sandbox https://etc33.codesandbox.io/. Examine the `nextEffect` pointer of `<Sidebar>` after clicking "Remount InnerBody" several times. You can see that it remains pointing at the original (and unmounted) `<InnerBody>`.

I tried to create this repro in the past but was bamboozled by `cloneChildFibers` and https://github.com/facebook/react/blob/67e3f3fb6e342f95f00215c84d5d013d7b0e1b33/packages/react-reconciler/src/ReactFiber.js#L418

Reply:
Maybe just add a comment next to it with a link to your repro. We don't need to add any infra just for this. We can figure that out later.

Reply:
@sebmarkbage in case it was missed, I opened #16115